src/modules/user/user.service.ts

```diff
@@ -30,4 +31,24 @@ export class UserService {
     this.users.push({ id, ...payload });
     return this.getUserById(id);
   }
+
+  updateUserById(id: number, payload: UpdateUserDto): UserDto {
+    this.users = this.getUsers().map((user) => {
```

Comment:
for me it's not clear what is going on in this method

Reply:
It's just doing some steps:
- get all users
- update current user by id which get from params
- return updated user

but it's work with global object which is mock data. This will removed in next pull request and instead this will use prisma client.